### PR TITLE
rollup: manage timestamps from L1 to L2 txs

### LINKED ISSUE #56: Timestamp Management for L1 to L2 Transactions
The timestamps are not being properly updated when transactions are ingested from L1 to L2 and causing errors when submitting batches. Update the logic so that this is fixed.

### PR DESCRIPTION
**Pull request by tynes**
## Description

This PR will update the blockchain time according the the timestamp on the block that includes the L1 to L2 transaction.

Closes https://github.com/ethereum-optimism/go-ethereum/issues/56

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](./CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: rollup/transaction_ingestion.go [hunk 1]
```diff
@@ -129,6 +131,11 @@ func (t *TxIngestion) loop() {
 		for i, tx := range txs {
 			log.Debug("Transaction Ingestion", "hash", tx.Hash().Hex(), "submission index", index, "element", i)
 
+			// Set the timestamp of the chain equal to the timestamp from the
+			// L1 to L2 transaction
+			timestamp := timestamps[i]
+			t.bc.SetCurrentTimestamp(int64(timestamp))
```
Comment (karlfloersch) [inline rollup/transaction_ingestion.go:137]: Just to make sure, we shouldn't have a race condition here at all right?

File: rollup/transaction_ingestion_sql.go [hunk 2]
```diff
@@ -98,11 +99,11 @@ func GetMostRecentQueuedTransactions(db *sqlx.DB) ([]*types.Transaction, uint32,
 	submissionIndex := submissionIndices[0]
 	for i := 1; i < len(submissionIndices); i++ {
 		if submissionIndices[i] != submissionIndex {
-			return nil, 0, fmt.Errorf("Submission index mismatch %d and %d", submissionIndex, submissionIndices[i])
+			return nil, 0, nil, fmt.Errorf("Submission index mismatch %d and %d", submissionIndex, submissionIndices[i])
 		}
 	}
 
-	return transactions, submissionIndex, nil
+	return transactions, submissionIndex, timestamps, nil
```
Comment (karlfloersch) [inline rollup/transaction_ingestion_sql.go:106]: Cool makes sense

**Review by karlfloersch — APPROVED** [review]
This looks great! I guess my only paranoid fear is could there be a race condition wrt setting the timestamp? I would be surprised if that was the case but figured we could think about it a little bit if you think that it is plausible.

File: rollup/transaction_ingestion.go (hunk 1 above)
Comment (karlfloersch) [reply rollup/transaction_ingestion.go:137]: We should add a check that makes sure the timestamp never goes down to this and the other timestamp update logic that we have

**Comment by tynes** [comment]
Closing this because of the stateless services

**Pull request closed without merging**